### PR TITLE
rpc: add `calculatenetworkfee` RPC method

### LINKED ISSUE #1842: calculatenetworkfee RPC method
We don't really need it, but at the same time it's a standard method that's not hard to implement: https://docs.neo.org/v3/docs/en-us/reference/rpc/latest-version/api/calculatenetworkfee.html

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1842.

TODO:

- [x] Add (* Client) CalculateNetworkFee
- [x] Investigate whether we can replace `(* Client) AddNetworkFee` by the `(* Client) CalculateNetworkFee`
- [x] Tests

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
>  Investigate whether we can replace (* Client) AddNetworkFee by the (* Client) CalculateNetworkFee
@roman-khimov, we can, but with a very dirty hack related to Notary contract, see explanation inside the commit. Is it appropriate for us?

File: pkg/rpc/server/server.go [hunk 1]
```diff
@@ -548,6 +550,86 @@ func (s *Server) validateAddress(reqParams request.Params) (interface{}, *respon
...
+		var verificationScript []byte
+		for _, w := range tx.Scripts {
+			if w.VerificationScript != nil && hash.Hash160(w.VerificationScript).Equals(signer.Account) {
+				// then it's a standard sig/multisig witness
+				verificationScript = w.VerificationScript
+				break
+			}
+		}
+		if verificationScript == nil { // then it still might be a contract-based verification
+			res, respErr := s.runScriptInVM(trigger.Verification, []byte{}, signer.Account, tx)
```
Comment (roman-khimov) [inline pkg/rpc/server/server.go:585]: Take invocation script instead of `[]byte{}`? It can be present in the transaction and some contract-based verifications can require things from invocation script.

File: pkg/rpc/server/server.go [hunk 2]
```diff
@@ -548,6 +550,86 @@ func (s *Server) validateAddress(reqParams request.Params) (interface{}, *respon
...
+			if respErr != nil && errors.Is(respErr.Cause, core.ErrUnknownVerificationContract) {
+				// it's neither a contract-based verification script nor a standard witness attached to
+				// the tx, so the user did not provide enough data to calculate fee for that witness =>
+				// we don't take it into account.
+				continue // TODO: @roman-khimov, I'd better return an error in this case, because
+				// 1) the user might expect that RPC node wallet contains the proper account with verification script (we don't have the wallet here => error)
+				// 2) the user might just forgot to add the proper verification script to the transaction witnesses (it's a user error => we need to notify about it)
+				// but C# code continues here, see https://github.com/neo-project/neo/blob/master/src/neo/Wallets/Wallet.cs#L551
```
Comment (roman-khimov) [inline pkg/rpc/server/server.go:593]: I'd file it as a bug for C# code.

File: pkg/rpc/client/rpc.go [hunk 3]
```diff
@@ -774,53 +775,40 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 }
 
 // AddNetworkFee adds network fee for each witness script and optional extra
-// network fee to transaction. `accs` is an array signer's accounts.
+// network fee to transaction. `accs` is an array of signer's accounts with
+// matching order.
```
Comment (roman-khimov) [inline pkg/rpc/client/rpc.go:779]: We better not touch it for now (move to separate PR for discussion).

File: pkg/rpc/server/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpc/server/server.go:585]: https://github.com/neo-project/neo/issues/2412

Comment (roman-khimov) [reply pkg/rpc/server/server.go:585]: It's OK to have this as an extension for now.

File: pkg/rpc/server/server.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/rpc/server/server.go:593]: https://github.com/neo-project/neo/pull/2413

File: pkg/rpc/server/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpc/server/server.go]: Implemented in a separate commit.

File: pkg/rpc/server/server.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/rpc/server/server.go]: Also implemented as proposed to Neo devs.

File: pkg/rpc/client/rpc.go (hunk 3 above)
Comment (AnnaShaleva) [reply pkg/rpc/client/rpc.go]: #1863.

**Review by roman-khimov — APPROVED** [review]
